numcodecs/bitround.py

```diff
@@ -29,19 +30,30 @@ class BitRound(Codec):
     Parameters
     ----------
 
-    keepbits: int
+    keepbits: int or function
         The number of bits of the mantissa to keep. The range allowed
         depends on the dtype input data. If keepbits is
         equal to the maximum allowed for the data type, this is equivalent
-        to no transform.
+        to no transform. Alternatively, pass a function to determine the
```

Comment:
Can we have an example of where such a bitinformation function may be found?

Reply:
Namely, [xbitinfo](https://github.com/observingClouds/xbitinfo), which was the impetus for BitRounding codec. More recently, that group has advocated for computing bitinformation chunk by chunk, using something like 
```python
fn = 'air.zarr'
ds.to_compressed_zar(fn, compute=False, mode='w')

dims = ds.air.dims
len_dims = len(dims)
slices = slices_from_chunks(ds.air.chunks)

for b, block in enumerate(ds.air.data.to_delayed().ravel()):
  ds_block = xr.Dataset({'air':(dims, block.compute())})
  rounded_ds = bitrounding(ds_block)
  rounded_ds.to_zarr(fn, region={dims[d]:s for (d,s) in enumerate(slices[b])})
```

Modifying the codec could make chunk-wise bitrounding workflows much simpler, and my proposed change would be the minimal first step.

Reply:
I was suggesting information like this should go into the docstring, so that users can act on it